### PR TITLE
Add TestCase tests and split TestCase mock methods into trait

### PR DESCRIPTION
**Pull request by unfulvio-godaddy**
# Summary 

This PR adds several test cases for the `TestCase` class. Also splits some methods mocking WordPress objects into own trait.

## Details

Other than splitting those mocking methods into a trait, there is no logic code change, although the `TestCase` has methods rearranged.

## Contributor checklist 

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly 
- [x] I have added tests to cover changes introduced by this pull request
- [x] All new and existing tests pass

## Testing 

- [x] PHPUnit tests pass

### Reviewer checklist 

 

- [x] Code changes review
- [ ] Documentation changes review
- [x] Unit tests pass
- [x] Static analysis passes

### CODE REVIEW

**Review by ajaynes-godaddy — APPROVED** [review]
Seems good here. Definitely a lot of churn in `TestCase.php` so it's a little hard to follow but I see the now-deprecated methods and everything seems in order.

**Review by agibson-godaddy — APPROVED** [review]
